### DISCUSSION TITLE
Apple TV support?

### DISCUSSION THREAD

**Discussion by pollocluck** [UnicornsOnLSD/finamp#897] [General]
Hello! I was wondering if there will be plans to make an Apple TV version for this app, since Swiftfin doesn't support music playback on Apple TV as of today.

**Comment by Chaphasilor** [comment]
Firstly, thanks foe creating a discussion instead of an issue. I should really start to convert some existing issues to discussions...

As for AppleTV support, it is a possibility. But there are a few problems:
1. The Flutter framework doesn't support AppleTV officially
2. Developing for Apple devices requires an Apple computer (which I don't have)
3. Finamp isn't yet optimized for a TV interface

I've dabbled with Android TV in the past, since the regular Finamp APK can already be installed there. But while most things work, there are still some rather severe bugs and navigation issues.

Building for AppleTV would be much more involved, but doable, as explained in this post: https://medium.com/flutter-community/flutter-for-apple-tv-756fcd5e8113

It requires building a custom Flutter engine that adds support for tvOS, as well as some additional logic for properly handling the inputs from an AppleTV remote.

So if anyone is willing to look into this, I'd be super happy about it, and any effort in making Finamp usable on an AppleTV should make is more compatible with Android TV at the same time, which would be great.  
But I can't help out with this myself, aside from eventually working on a proper TV layout. But that's not a priority at the moment.

**Comment by pollocluck** [comment]
No problem and thanks for responding! I didn't realize a Mac would be needed to create a tvOS app. I understand this may not be a top priority for the contributors at the moment, but it's a little disappointing that the fire stick and Roku versions have support for more media types than the Apple/Android tv clients.

**Comment by Chaphasilor** [comment]
I believe FireTV and Android TV are essentially the same platform, they both run Android apps. The Roku app is/was maintained by @1hitsong who is a Finamp contributor and an artist themselves, which probably explains the dedication.

Swiftfin started out as an inofficial client driven by community contributors, but it seems like no one was interested in adding music support. It's a lot of work after all, and pretty different from video when it comes to the UI.

In the end, adding support for major new features (like Finamp supporting tvOS or Swiftfin supporting music) usually stand and falls with people contributing it themselves, or at least kicking things off.

**Reply by 1hitsong** [reply]
> The Roku app is/was maintained by @1hitsong who is a Finamp contributor and an artist themselves, which probably explains the dedication.

I'm no longer on the official Jellyfin Roku team, I'm now working on my own fork of the Roku client.

**Reply by Chaphasilor** [reply]
Yeah, I took a closer look afterwards and saw viv ^^
That's why included "was".

Either way, you're probably at least part of the reason why the official client had so many music-related features 😁
